packages/suite/src/actions/wallet/transactionActions.ts

```diff
@@ -1,72 +1,166 @@
+//         dispatch({
+//             type: TRANSACTION.UPDATE,
+//             txId,
+//             timestamp: updatedTimestamp,
+//         });
+//     });
+// };
+
+const getTransactionsFromStorage = async (descriptor: string, offset?: number, count?: number) => {
     try {
-        await db.addItem('txs', transaction).then(_key => {
-            dispatch({
-                type: TRANSACTION.ADD,
-                transaction,
-            });
+        const txs = await db.getItemsExtended('txs', 'accountId-blockTime', {
+            key: descriptor,
+            offset,
+            count,
         });
+        return txs;
     } catch (error) {
-        if (error && error.name === 'ConstraintError') {
-            console.log('Tx with such id already exists');
-        } else if (error) {
-            console.error(error.name);
-            console.error(error.message);
-        } else {
-            console.error(error);
-        }
+        console.log(error);
+        return null;
     }
 };
 
-export const remove = (txId: string) => async (dispatch: Dispatch) => {
-    db.removeItem('txs', 'txId', txId).then(() => {
-        dispatch({
-            type: TRANSACTION.REMOVE,
-            txId,
-        });
-    });
-};
+export const fetchTransactions = (account: Account, page: number, perPage?: number) => async (
+    dispatch: Dispatch,
+    getState: GetState,
+) => {
+    const { selectedAccount } = getState().wallet;
+    // const totalTxs = selectedAccount.account!.history.total;
+    const { transactions } = getState().wallet.transactions;
+    const reducerTxs = getAccountTransactions(transactions, account);
+    const txsForPage = reducerTxs.filter(t => t.page === page);
+    // we already got txs for the page in reducer
+    if (txsForPage.length > 0) return;
 
-export const update = (txId: string) => async (dispatch: Dispatch) => {
-    const updatedTimestamp = Date.now();
-    db.updateItemByIndex('txs', 'txId', txId, { timestamp: updatedTimestamp }).then(_key => {
-        dispatch({
-            type: TRANSACTION.UPDATE,
-            txId,
-            timestamp: updatedTimestamp,
-        });
+    dispatch({
+        type: TRANSACTION.FETCH_INIT,
     });
-};
 
-export const getFromStorage = (accountId: number, offset?: number, count?: number) => async (
-    dispatch: Dispatch,
-): Promise<void> => {
-    db.getItemsExtended('txs', 'accountId-blockTime', {
-        key: accountId,
-        offset,
-        count,
-    }).then((transactions: WalletAccountTransaction[]) => {
+    // TODO: storing and fetching from the db
+    const offset = page && perPage ? (page - 1) * perPage : undefined;
+    const count = perPage || undefined;
+    const storedTxs = await getTransactionsFromStorage(account.descriptor, offset, count);
+
+    const shouldFetchFromBackend = storedTxs === null || storedTxs.length === 0;
+    if (shouldFetchFromBackend) {
+        let result: ResponseMessage<AccountInfo> | null = null;
```

Comment:
use `ReturnType<typeof TrezorConnect.getAccountInfo>` instead of `ResponseMessage`